File: backend/ViewModel/RecentActivityItem.cs
using System;
using System.Collections.Generic;

namespace tradeus.ViewModel
{
    public class RecentActivityItem
    {
        public List<RecentItem> RecentItems { get; set; }
        public bool ViewedRecentNotifications { get; set; }
    }

    public class RecentItem
    {
        public string Username { get; set; }
        public int PostId { get; set; }
        public RecentActivityItemType RecentActivityItemType { get; set; }
        public DateTime Created { get; set; }
    }

    public enum RecentActivityItemType
    {
        Follow, PostLike, CommentLike
    }
}

File: frontend/src/models/RecentActivityItem.ts
export class RecentActivityItem {
    public recentItems: RecentItem[] = [];
    public viewedRecentNotifications = true;
}

export class RecentItem {
    public username = "";
    public postId = 0;
    public created = new Date()
    public recentActivityItemType: RecentActivityItemType = 0;

}

export enum RecentActivityItemType {
    Follow, PostLike, CommentLike
}
